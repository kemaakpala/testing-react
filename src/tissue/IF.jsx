
const IF = (props) => {
  const {condition, children} = props;
  return (condition) ? children : null;
} ;

export default IF;