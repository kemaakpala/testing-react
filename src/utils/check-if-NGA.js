const isNGA = () => false;

export { isNGA };