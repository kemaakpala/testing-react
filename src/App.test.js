import React from 'react';
import {
  render,
  cleanup
} from 'react-testing-library';
import 'jest-dom/extend-expect';
import userEvent from 'user-event';
import IF from './tissue/IF';
import SelectFormField from './atom/select-form-field';
import PostCodeFormField from './atom/post-code-form-field';
import DurationInputFormField from './atom/duration-input-form-field';
import { hashCode } from './utils';
import optionsData from './refData/data';

afterEach(cleanup);

it('renders children if condition is true', () => {
  const isTrue = true;
  const { asFragment } = render(
    <div>
      <IF condition={isTrue}>
        {'testing'}
      </IF>
    </div>
    
  );
  expect(asFragment()).toMatchSnapshot();
});

it('renders a select field from props', () => {
  const optData = optionsData();
  const { asFragment } = render(
    <SelectFormField 
      label="Residential Status"
      name="residentialStatus"
      options={optData}
      placeholder={"Please Select"}
    />
  );
  expect(asFragment()).toMatchSnapshot();
});

it('renders a postCode Form Field', () => {
  const onChange = jest.fn();
  const { getByTestId } = render(
    <PostCodeFormField
      label="PostCode"
      name="currentAddress"
      onChange={onChange}
      value="HA4 0GD"
    />
  );
  userEvent.type(getByTestId("postCodeField"), "HA4 0GD");
  expect(getByTestId("postCodeField")).toMatchSnapshot();
  expect(getByTestId("postCodeField")).toHaveAttribute("value", "HA4 0GD");
});

it('it returns a number hash', () => {
  const str = 'testString';
  const hashString = hashCode(str);
  expect(hashString).toEqual(1727748931);
});

it('matches snapshot of duration fields', ()=> {
  const onChange = jest.fn();
  const newDate = new Date();
  const startYear = 1965;
  const endYear = newDate.getFullYear();
  const duration = endYear - startYear;
  const { getByTestId} = render(
    <DurationInputFormField
      label="When did you move here?"
      name="durationOfStay"
      order="descending"
      durationText="Time at this address"
      startYear={startYear}
      endYear={endYear}
      value={duration}
      onChange={onChange}
    />
  );

  expect(getByTestId("startYear")).toMatchSnapshot();
  expect(getByTestId("endYear")).toMatchSnapshot();
});

test('to Contain Element', () => {
  const onChange = jest.fn();
  const newDate = new Date();
  const startYear = 1965;
  const endYear = newDate.getFullYear();
  const duration = endYear - startYear;

  const { asFragment } = render(
    <DurationInputFormField
      label="When did you move here?"
      name="durationOfStay"
      order="descending"
      durationText="Time at this address"
      startYear={startYear}
      endYear={endYear}
      value={duration}
      onChange={onChange}
    />
  );

  expect(asFragment()).toMatchSnapshot();
});
