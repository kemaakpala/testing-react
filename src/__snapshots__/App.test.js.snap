// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot of duration fields 1`] = `
<input
  data-testid="startYear"
  id="startYear"
  name="durationOfStay"
  type="text"
  value="1965"
/>
`;

exports[`matches snapshot of duration fields 2`] = `
<input
  data-testid="endYear"
  id="endYear"
  name="durationOfStay"
  type="text"
  value="2019"
/>
`;

exports[`renders a postCode Form Field 1`] = `
<input
  data-testid="postCodeField"
  name="currentAddress"
  type="text"
  value="HA4 0GD"
/>
`;

exports[`renders a select field from props 1`] = `
<DocumentFragment>
  <label
    for="residentialStatus"
  >
    Residential Status
  </label>
  <select
    id="residentialStatus"
    name="residentialStatus"
  >
    <option
      value=""
    >
      Please Select
    </option>
    <option
      value="1"
    >
      resident
    </option>
    <option
      value="2"
    >
      A resident not ordnarily resident (RNOR)
    </option>
    <option
      value="3"
    >
      A non-resident (NR)
    </option>
  </select>
</DocumentFragment>
`;

exports[`renders children if condition is true 1`] = `
<DocumentFragment>
  <div>
    testing
  </div>
</DocumentFragment>
`;

exports[`to Contain Element 1`] = `
<DocumentFragment>
  <div>
    When did you move here?
  </div>
  <p>
    <span />
  </p>
  <label
    for="startYear"
  />
  <input
    data-testid="startYear"
    id="startYear"
    name="durationOfStay"
    type="text"
    value="1965"
  />
  <label
    for="endYear"
  />
  <input
    data-testid="endYear"
    id="endYear"
    name="durationOfStay"
    type="text"
    value="2019"
  />
  <div
    data-testid="parent"
  >
    <span
      data-id="child"
    >
      54
    </span>
  </div>
</DocumentFragment>
`;
